File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string ans;
        int l1=word1.length();
        int l2=word2.length();
        int i = 0;
        int itr1=0;
        int itr2=0;
        while(itr1<l1 || itr2<l2){            
            if(i%2==0 && itr1 < l1){
                // add first string
                ans.push_back(word1[itr1++]);
            }
            if(i%2!=0 && itr2 < l2){
                // add second string
                ans.push_back(word2[itr2++]);
            }          
            i++;
        }
        if(itr1!=l1-1){
            while(itr1<l1) ans.push_back(word1[itr1++]);
        }
        if(itr2!=l2-1){
            while(itr2<l2) ans.push_back(word2[itr2++]);
        }
        return ans;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut ans: String = String::from("");
        let mut len1: usize = 0;
        let mut len2: usize = 0;
        // The solution is much more simpler than I thought
        // Time Complexity : O(n) [n is max of len1 and len2]
        while len1 < word1.len() || len2 < word2.len() {
            if len1 < word1.len() {
                ans.push(word1.chars().nth(len1).unwrap());
                len1 += 1;
            }
            if len2 < word2.len() {
                ans.push(word2.chars().nth(len2).unwrap());
                len2 += 1;
            }
        }
        
        ans
    }
}
